features/FEATURE_BLE/targets/TARGET_CORDIO/TESTS/cordio_hci/transport/main.cpp

```
@@ -0,0 +1,190 @@
+        switch (state) {
+            case WAITING_EVENT_PACKET:
+                if (data[i] == HCI_EVT_TYPE) {
+                    state = WAITING_EVENT_CODE_COMPLETE;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+
+            case WAITING_EVENT_CODE_COMPLETE:
+                if (data[i] == HCI_CMD_CMPL_EVT) {
+                    state = WAITING_PARAMETER_LENGTH;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+
+            case WAITING_PARAMETER_LENGTH:
+                if (data[i] == RESET_PARAMETER_LENGTH) {
+                    state = WAITING_NUM_HCI_EVT_PACKET;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+
+            case WAITING_NUM_HCI_EVT_PACKET:
+                // controler dependent; can be any value, pass on to the next token
+                state = WAITING_OPCODE_LSB;
+                break;
+
+            case WAITING_OPCODE_LSB:
+                if (data[i] == HCI_OPCODE_RESET_LSB) {
+                    state = WAITING_OPCODE_MSB;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+
+            case WAITING_OPCODE_MSB:
+                if (data[i] == HCI_OPCODE_RESET_MSB) {
+                    state = WAITING_STATUS;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+
+            case WAITING_STATUS:
+                if (data[i] == RESET_EXPECTED_STATUS) {
+                    test_result = TEST_RESULT_SUCCESS;
+                    state = DONE;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+        }
+
+        if (test_result != TEST_RESULT_TIMEOUT_FAILURE) {
```

Presumably that should be outside of the for loop?

---

Why ? there's no guarantee that data in input is at the event boundary and it may fail early.